ark/analysis/spatial_analysis.py

```diff
@@ -295,3 +295,54 @@ def create_neighborhood_matrix(all_data, dist_matrices_dict, included_fovs=None,
         cell_neighbor_freqs.loc[current_fov_neighborhood_data.index, fov_cluster_names] = freqs
 
     return cell_neighbor_counts, cell_neighbor_freqs
+
+
+def compute_cluster_metrics(neighbor_mat, max_k=10, included_fovs=None,
+                            fov_col='SampleID', label_col='cellLabelInImage'):
+    """Given a neighborhood matrix, provide a process to produce k-means clustering metrics.
```

Comment:
For docstrings, the first line should always be a single line summary, followed by a blank space. This should be a self-contained one line summary of the function. If more description is required, that can follow the initial one line. But the first line should be the brief, standalone description.